app/assets/javascripts/shared/behaviors.js

```diff
@@ -6,4 +6,27 @@ document.addEventListener('turbolinks:load', function(){
   $('[data-behavior~=rich-toolbar]').each(function() {
     new EditorToolbar($(this));
   });
-})
+
+  // Activate Modals for images within textile rendered areas
+  // The ImageModal class will control the navigation of the image carousel. In
+  // this case the modal is managed by bootstrap. Bootstrap brings the modal up
+  // on click, we access the needed images as some may have been ajaxed in
+  // (comments, or the editor preview) and add the images to the ImageModal
+  // object to be displayed within the modal.
+  const $modal = $('[data-behavior~=image-modal]');
+  const targetImgs = 'img[data-toggle=modal]';
+  var modal;
+
+  $('[data-behavior~=view-content]').on('click', targetImgs, function() {
+    if (modal !== undefined) { modal.unbind(); }
+    modal = new ImageModal([], 0, $modal);
+
+    scope = ($(this).parents('[data-behavior~=comment-feed]').length ? 'comment-feed' : 'content-textile');
+    $(this).parents('[data-behavior~='+scope+']').find(targetImgs).each(function() {
+      modal.addImage(this);
+    });
+
+    var index = modal.getIndex(this);
+    modal.loadImage(index);
+  });
```

Comment:
I'm not sure of our architecture tbh.

Click to enlarge? Cool!

Do we really need a slideshow of all the screenshots in a given stream? Not so sure.

If I click on the screenshot on a comment from Alice, I don't think I care about being able to navigate to a screenshot 7 comment up posted by Bob using the chevron. It seems we went too far for a v1 of this feature.

From a technical perspective:

- We're installing a new `click` handler for every image.
- However in each handler, we're reusing a global variable (`modal`) that we `unbind()` and reinistantiate every single time.
- But this `ImageModal` instance *always* does the same in the same page: load all images available. If we click on 3 images, we unbind and reinstantiate 3 times, doing exactly the same every time.
- L24 is a smell, `data-behaviors` can be combined, here we don't need to choose which one is on the page, we need to add a *new* data-behavior for this *new* behavior we're adding.
- L 25 also feels out of place, we're already looking for all `targetImgs` in L20, then we're in a single image, click handler context, but we do so again. Seems to me that the `ImageModal` init and `addImage` don't belong to the click handler. But as I already mentioned, I don't think we need the extra bells and whistles at all.